DEMO-37

Anii555 opened the pull request.

Anii555 committed:
В верном направлении двигаюсь?

Пропсы же ещё нужны, да?

Corel-frim commented on src/App.vue:
```diff
@@ -198,9 +199,7 @@ export default {
 
   methods: {
     addInBasket(id) {
-      axios.post(`http://localhost:5090/cart/${id}`).then(() => {
```
В апи должен уйти только запрос, т.е. axios.post(...), а then и пр. остаться тут

Anii555 replied:
✅

Anii555 replied:
✅

Corel-frim commented on src/App.vue:
```diff
@@ -214,17 +213,7 @@ export default {
   mounted: function () {
     //вывод из бд
-    axios
-      .get("http://localhost:5090/product/getAllProducts")
```
Вот эти 2 строки должны уйти в апи, все что ниже - остаться

Anii555 replied:
✅

Corel-frim commented on src/cartApi.js:
```diff
@@ -0,0 +1,7 @@
+const axios = require("axios").default;
+
+function addInBasket(id) {
+  axios.post(`http://localhost:5090/cart/${id}`).then(() => {
```
return не стоит забывать, поскольку это асинхронный запрос, они в js построены на промисах (типа как Task в шарпе)
И должно остаться только axios.post(...)
Имеет смысл вынести `http://localhost:5090/cart/` в константу в апи, поскольку внутри апи оно должно использоваться несколько раз

Anii555 replied:
✅

Corel-frim commented on src/cartApi.js:
```diff
@@ -0,0 +1,7 @@
+const axios = require("axios").default;
```
Переделать на import

Anii555 replied:
✅

Corel-frim commented on src/productsApi.js:
```diff
@@ -0,0 +1,16 @@
+const axios = require("axios").default;
+
+mounted: function getAllProducts() {
```
mounted тут быть не должно, в остальном замечания те же, что и к cartApi

Anii555 replied:
✅

Corel-frim commented on src/productsApi.js:
```diff
@@ -0,0 +1,16 @@
+      console.error(e);
+    });
+
+  this.$root.$on(`cart_sum`, this.changeSum);
```
Эта строка не нужна вообще в проекте, т.к. уже используется @cart_sum

Anii555 replied:
✅

Corel-frim reviewed (changes requested):
А CartProducts.vue?

Anii555 committed:
Сделаль, НО

Module not found: Error: Can't resolve '/cartApi.js' in

App не видит мои .js: 'C:\Users\13ann\OneDrive\Документы\JS Projects\API_OnlineShop_frontend\src'
Экспортирую .js

Corel-frim commented on src/components/CartProducts.vue:
```diff
@@ -126,7 +126,7 @@
 
 <script>
-import axios from "axios";
+import cartApi from "./API_OnlineShop_frontend/src/cartApi.js";
```
А разве не ../cartApi?

Anii555 replied:
✅

Corel-frim commented on src/cartApi.js:
```diff
@@ -0,0 +1,24 @@
+const path_cart = `http://localhost:5090/cart/`;
+
+export function addInBasket(id) {
+  return axios.post(path_cart + `${id}`);
```
Можно просто path_cart + id

Anii555 replied:
✅

Corel-frim commented on src/cartApi.js:
```diff
@@ -0,0 +1,24 @@
+}
+
+export function delCartItem(amount, id) {
+  return axios.delete(path_cart + `${id}`);
```
Можно просто path_cart + id

Anii555 replied:
✅

Corel-frim commented on src/cartApi.js:
```diff
@@ -0,0 +1,24 @@
+  return axios.post(path_cart + `${id}`);
+}
+
+export function updateCart() {
```
Так это же не обновление

Anii555 replied:
✅

Corel-frim commented on src/components/CartProducts.vue:
```diff
@@ -152,8 +152,8 @@ export default {
+        .updateCart()
         .then((resp_cart) => {
           console.log("проверыч корзины: ");
           console.log(resp_cart.data);
```
И удалить все записи в консоль не забудь

Anii555 replied:
✅

Anii555 committed:
Фикс правок
